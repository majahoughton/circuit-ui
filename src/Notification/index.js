import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import withStyles from '../../util/withStyles';
import styles from './index.scss';

const Notification = ({ type, msg, children, className }) => {
  if (!(msg || children)) return null;
  const content = children || msg;
  const classes = classNames(
    {
      alert: true
    },
    `alert--${type}`,
    className
  );
  return <div className={classes}>{content}</div>;
};

Notification.propTypes = {
  type: PropTypes.string,
  data: PropTypes.object,
  children: PropTypes.any
};

export default withStyles(styles)(Notification);
