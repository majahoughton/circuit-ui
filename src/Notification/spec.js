import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Notification from '.';

describe('Notification', () => {
  it('should render error type when error prop is passed', () => {
    const notification = renderer.create(
      <Notification type="error" msg="there was an error" />
    );
    expect(notification).toMatchSnapshot();
  });
  it('should not render if there happens to be no children or data', () => {
    const notification = renderer.create(<Notification />);
    expect(notification).toMatchSnapshot();
  });
  it('should take the body text as a child', () => {
    const output = mount(<Notification>Hello World</Notification>);
    expect(output.text()).toContain('Hello World');
  });
});
