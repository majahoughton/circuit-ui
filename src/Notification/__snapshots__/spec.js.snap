// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should not render if there happens to be no children or data 1`] = `null`;

exports[`Notification should render error type when error prop is passed 1`] = `
<div
  className="alert alert--error"
>
  there was an error
</div>
`;
